tests/integ/test_data_source_api.py

```diff
@@ -348,3 +367,55 @@ def assert_datasource_statement_params_run_query(*args, **kwargs):
             single=True,
         )
         assert called == 2
+
+
+@pytest.mark.skipif(
+    IS_WINDOWS,
+    reason="sqlite3 file can not be shared accorss processes on windows",
+)
+@pytest.mark.parametrize(
+    "custom_schema",
+    [
+        "id INTEGER, int_col INTEGER, real_col FLOAT, text_col STRING, blob_col BINARY, null_col STRING, ts_col TIMESTAMP, date_col DATE, time_col TIME, short_col SHORT, long_col LONG, double_col DOUBLE, decimal_col DECIMAL, map_col MAP, array_col ARRAY, var_col VARIANT",
+        StructType(
+            [
+                StructField("id", IntegerType()),
+                StructField("int_col", IntegerType()),
+                StructField("real_col", FloatType()),
+                StructField("text_col", StringType()),
+                StructField("blob_col", BinaryType()),
+                StructField("null_col", NullType()),
+                StructField("ts_col", TimestampType()),
+                StructField("date_col", DateType()),
+                StructField("time_col", TimeType()),
+                StructField("short_col", ShortType()),
+                StructField("long_col", LongType()),
+                StructField("double_col", DoubleType()),
+                StructField("decimal_col", DecimalType()),
+                StructField("map_col", MapType()),
+                StructField("array_col", ArrayType()),
+                StructField("var_col", VariantType()),
+            ]
+        ),
+    ],
+)
+def test_custom_schema(session, custom_schema):
```

Comment:
is there a reason that we only test custom schema in sqllite3 while not test sql server?

Reply:
this is to test the functionality of setting custom schema for the target snowflake db, the source database doesn't matter as long as we have the data prepared.

on the other hand, using sqlite3 because it's easy to set up, it's a python built-in lib using file as the storage. on the other hand, sql server requires docker image / or secret management which introduces extra cycles to dev